test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1121,53 +1079,205 @@ UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup, compare_cm_and_wireup_configs,
+UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup_fail,
                      connect_and_fail_wireup_msg_pack_addr_on_client,
                      !cm_use_all_devices())
 {
-    connect_and_fail_wireup(sender(), UCP_EP_FLAG_CLIENT_CONNECT_CB,
-                            FAIL_WIREUP_MSG_ADDR_PACK);
+    connect_and_fail_wireup(sender(), FAIL_WIREUP_MSG_ADDR_PACK,
+                            UCP_EP_FLAG_CLIENT_CONNECT_CB);
 }
 
-UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup,
+UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup_fail,
                      connect_and_fail_wireup_msg_pack_addr_on_server,
                      !cm_use_all_devices())
 {
-    connect_and_fail_wireup(receiver(), UCP_EP_FLAG_SERVER_NOTIFY_CB,
-                            FAIL_WIREUP_MSG_ADDR_PACK);
+    connect_and_fail_wireup(receiver(), FAIL_WIREUP_MSG_ADDR_PACK,
+                            UCP_EP_FLAG_SERVER_NOTIFY_CB);
 }
 
-UCS_TEST_P(test_ucp_sockaddr_wireup,
+UCS_TEST_P(test_ucp_sockaddr_wireup_fail,
            connect_and_fail_wireup_set_ep_failed_on_client)
 {
-    connect_and_fail_wireup(sender(), UCP_EP_FLAG_CLIENT_CONNECT_CB,
-                            FAIL_WIREUP_SET_EP_FAILED);
+    connect_and_fail_wireup(sender(), FAIL_WIREUP_SET_EP_FAILED,
+                            UCP_EP_FLAG_CLIENT_CONNECT_CB);
 }
 
-UCS_TEST_P(test_ucp_sockaddr_wireup,
+UCS_TEST_P(test_ucp_sockaddr_wireup_fail,
            connect_and_fail_wireup_set_ep_failed_on_server)
 {
-    connect_and_fail_wireup(receiver(), UCP_EP_FLAG_SERVER_NOTIFY_CB,
-                            FAIL_WIREUP_SET_EP_FAILED);
+    connect_and_fail_wireup(receiver(), FAIL_WIREUP_SET_EP_FAILED,
+                            UCP_EP_FLAG_SERVER_NOTIFY_CB);
 }
 
-UCP_INSTANTIATE_ALL_TEST_CASE(test_ucp_sockaddr_wireup)
+UCP_INSTANTIATE_ALL_TEST_CASE(test_ucp_sockaddr_wireup_fail)
+
+
+class test_ucp_sockaddr_wireup_fail_try_next_cm :
+        public test_ucp_sockaddr_wireup_fail  {
+public:
+    test_ucp_sockaddr_wireup_fail_try_next_cm()
+    {
+        m_try_next_cm_cb_found = false;
+    }
+
+private:
+    static int find_try_next_cm_cb(const ucs_callbackq_elem_t *elem, void *arg)
+    {
+        if ((elem->cb == ucp_cm_client_try_next_cm_progress) &&
+            (elem->arg == arg)) {
+            m_try_next_cm_cb_found = true;
+        }
+
+        return 0;
+    }
+
+    virtual bool wait(entity &e, uint32_t wait_ep_flags)
+    {
+        if (test_ucp_sockaddr_wireup_fail::wait(e, wait_ep_flags)) {
+            UCS_TEST_SKIP_R("trying the next CM calback wasn't scheduled");
+        }
+
+        UCS_ASYNC_BLOCK(&e.worker()->async);
+        uct_priv_worker_t *worker = ucs_derived_of(e.worker()->uct,
+                                                   uct_priv_worker_t);
+        ucs_callbackq_remove_if(&worker->super.progress_q,
+                                find_try_next_cm_cb, e.ep());
+        UCS_ASYNC_UNBLOCK(&e.worker()->async);
+
+        if (m_try_next_cm_cb_found) {
```

Comment:
pass custom arg to remove_if instead of using global var m_try...
e.g
bool is_found=false;
remove_if(..., &is_found);

Reply:
done